doc/guides/primordials.md

```diff
@@ -0,0 +1,294 @@
+# Usage of primordials in core
+
+The file `lib/internal/per_context/primordials.js` subclasses and stores the JS
+built-ins that come from the VM so that Node.js built-in modules do not need to
+later look these up from the global proxy, which can be mutated by users.
+
+Usage of primordials should be preferred for any new code, but replacing current
+code with primordials should be done with care.
```

Comment:
This should link to https://github.com/nodejs/node/pull/38635/files#diff-1fd7756e2222fdc2c8dbd60f718237a78233a8697d418ec9044e167b673bcf96R94.